src/Package/Impl/Logging/DiagnosticLogs.cs

```diff
@@ -74,7 +75,7 @@ internal static class DiagnosticLogs {
             return zipPath;
         }
 
-        private static void CollectRTVSLogs(object o) {
+        private static void CollectRTVSLogs(object o, CancellationToken ct) {
```

Comment:
Actions may or may not be async but token is still useful. LongOperationNotification (as it is derived from the way VS does long cancellable ops) only checks for the cancellation between actions. So if there is only one action or one of the actions is long, they should be checking for cancellation periodically and exit when appropriate.